lib/LLVMIRCodeGen/LLVMBackend.cpp

```diff
@@ -99,6 +99,8 @@ void LLVMBackend::emitJitMain(LLVMIRGen &irgen) const {
   irgen.createCall(builder, entryF, initFunctionCallArgs);
   // Terminate the function.
   builder.CreateRetVoid();
+  // Emit JIT file printer.
+  irgen.generateJITFileWriter(func);
```

Comment:
LLVMBackend is used sometimes for AOT compilation, not just for a JIT. Thus, registering the JITFileWriter unconditionally could be not such a good idea in general. But I don't have a strong opinion here.

Reply:
I think of this "registering" as simply "providing". Backend is free not to use it. 
Note that we need CPU backend provided dumps as much as Xtensa dumps, throughout our compile and use flow. Therefore I moved registering here. We originally had a backend option to enabled this, however, we settled with the current approach merely due to it being one-liner, which allows for easier rebase until (and if ever) we upstream.